jbehave-support-core/src/main/java/org/jbehavesupport/core/rest/RestServiceHandler.java

```diff
@@ -461,12 +461,38 @@ public void verifyResponse(String expectedStatus, ExamplesTable data) {
         if (!StringUtils.isEmpty(expectedStatus)) {
             verifyResponseStatus(actualResponseStatus, parseHttpStatus(expectedStatus), actualResponseMessage);
         }
-        if (ExamplesTableUtil.tableContains(data, NAME, STATUS_HEADER::equals)) {
-            verifyResponseStatus(actualResponseStatus, parseHttpStatus(getValue(data, NAME, STATUS_HEADER, EXPECTED_VALUE)), actualResponseMessage);
+        if (ExamplesTableUtil.tableContains(convertedData, NAME, STATUS_HEADER::equals)) {
+            verifyResponseStatus(actualResponseStatus, parseHttpStatus(getValue(convertedData, NAME, STATUS_HEADER, EXPECTED_VALUE)), actualResponseMessage);
         }
 
-        verifyResponseHeaders(actualHeaders, data, actualResponseMessage);
-        verifyResponseJson(actualResponseBody, data, actualResponseMessage);
+        verifyResponseHeaders(actualHeaders, convertedData, actualResponseMessage);
+        verifyResponseJson(actualResponseBody, convertedData, actualResponseMessage);
+    }
+
+    private ExamplesTable convertCollectionNotation(ExamplesTable data) {
+        if (data != null && !data.getRows().isEmpty()) {
+            List<Map<String, String>> newMapList = data.getRows().stream()
+                .map(map -> convertIfNeeded(map))
+                .collect(Collectors.toList());
+            return data.withRows(newMapList);
+        } else {
+            return data;
+        }
+    }
+
+    private Map<String, String> convertIfNeeded(Map<String, String> map) {
+        String name = map.get(ExampleTableConstraints.NAME);
+        Matcher mainMatcher = Pattern.compile(PERIOD_REGEX).matcher(name);
+        while (mainMatcher.matches()) {
+            String list = mainMatcher.group(1);
+            String index = mainMatcher.group(2);
+            String member = mainMatcher.group(3);
+            index = index.replaceFirst("\\.", "[") + "]";
+            String newName = list + index + member;
```

Comment:
What about idea, that we can replacing via regexp, rather than string concatenation?

Reply:
I tried that and it didn't work - it fought with multiple occurences of the index and if I tried to do it one by one, the cycle looped forever. Does it have any significant advantages to use regex?

Reply:
I tried and it did...
```
    private Map<String, String> convertIfNeeded(Map<String, String> map) {
        String name = map.get(ExampleTableConstraints.NAME);
        String newName = name.replaceAll("\\.(\\d+)(\\.){0,1}", "\\[$1\\]$2");
        map.put(ExampleTableConstraints.NAME, newName);
        return map;
    }
```
As of advantages you are already using regex, and VM has pretty hard time to process all the matching and stuff .  After it did it, you make trivial concatenation... It is like I would ask you to bring me single reeses directly from US...